pages/topics/rpc.mdx

```diff
@@ -0,0 +1,425 @@
+
+###### Handshake Structure
+
+| Field     | Type    | Description                                                       |
+| --------- | ------- | ----------------------------------------------------------------- |
+| v         | integer | The RPC version, currently only `1`                               |
+| client_id | string  | The client ID of the Discord application connecting to the socket |
+
+###### Example Ready Event
+
+```json
+{
+  "cmd": "DISPATCH",
+  "data": {
+    "v": 1,
+    "config": {
+      "cdn_host": "cdn.discordapp.com",
+      "api_endpoint": "//canary.discord.com/api",
+      "environment": "production"
+    },
+    "user": {
+      "id": "246877849162743818",
+      "username": "jay_taelien",
+      "discriminator": "0",
+      "global_name": "Jay",
+      "avatar": "91b7bc37e924f78625f7ea582fdbac5d",
+      "bot": false,
+      "flags": 16544,
+      "premium_type": 2
+    }
+  },
+  "evt": "READY",
+  "nonce": null
+}
+```
+
+###### RPC Authorize Example
+
+```json
+{
+  "nonce": "f48f6176-4afb-4c03-b1b8-d960861f5216",
+  "args": {
+    "client_id": "192741864418312192",
+    "scopes": ["rpc", "identify"]
+  },
+  "cmd": "AUTHORIZE"
+}
+```
+
+###### RPC Authenticate Example
+
+```json
+{
+  "nonce": "5bb10a43-1fdc-4391-9512-0c8f4aa203d4",
+  "args": {
+    "access_token": "CZhtkLDpNYXgPH9Ml6shqh2OwykChw"
+  },
+  "cmd": "AUTHENTICATE"
+}
+```
+
+### Packet Payloads
+
+Packet payloads are JSON objects containing the command you want to send to the Discord client. Incoming packets will need to be correctly parsed to obtain the JSON response.
+
+###### Outgoing Payload Structure
+
+| Field    | Type   | Description                                                                                                           |
+| -------- | ------ | --------------------------------------------------------------------------------------------------------------------- |
+| cmd      | string | The [command](#rpc-commands) indicating the action of the request                                                     |
+| args     | object | The arguments coinciding with the [command](#rpc-commands) of the request                                             |
+| nonce    | string | A unique identifier (UUID) given to a [command](#rpc-commands) that will be echoed back to you upon a successful send |
+| evt? ^1^ | string | The event the app is subscribing to                                                                                   |
+
+^1^ Only required when `cmd` is `SUBSCRIBE` or `UNSUBSCRIBE`.
+
+<Alert type="info">
+
+The Discord IPC will echo back every command you send as a response. Use this as a lock-step feature to avoid flooding
+messages. This can be used to validate messages such as the Presence or Subscribes.
```

Comment:
```suggestion
The Discord IPC will echo back every command you send as a response. This can be used as a lock-step feature to avoid flooding
the gateway with messages. It can also be used to validate messages such as the Presence or Subscribes.
```